src/Service/SimXMLToZGWService.php

```diff
@@ -323,6 +334,161 @@ public function unescapeEigenschappen(array $zaakArray): array
     }//end unescapeEigenschappen()
 
 
+    /**
+     * Generates a download endpoint from the id of an 'Enkelvoudig Informatie Object' and the endpoint for downloads.
+     *
+     * @param string   $id               The id of the Enkelvoudig Informatie Object.
+     * @param Endpoint $downloadEndpoint The endpoint for downloads.
+     *
+     * @return string The endpoint to download the document from.
+     */
+    private function generateDownloadEndpoint(string $id, Endpoint $downloadEndpoint): string
+    {
+        // Unset the last / from the app_url.
+        $baseUrl = rtrim($this->parameterBag->get('app_url'), '/');
+
+        $pathArray = $downloadEndpoint->getPath();
+        foreach ($pathArray as $key => $value) {
+            if ($value == 'id' || $value == '[id]' || $value == '{id}') {
+                $pathArray[$key] = $id;
+            }
+        }
+
+        return $baseUrl.'/api/'.implode('/', $pathArray);
+
+    }//end generateDownloadEndpoint()
+
+
+    /**
+     * Creates or updates a file associated with a given ObjectEntity instance.
+     *
+     * This method handles the logic for creating or updating a file based on
+     * provided data. If an existing file is associated with the ObjectEntity,
+     * it updates the file's properties; otherwise, it creates a new file.
+     * It also sets the response data based on the method used (POST or other)
+     * and if the `$setResponse` parameter is set to `true`.
+     *
+     * @param ObjectEntity $objectEntity The object entity associated with the file.
+     * @param array        $data         Data associated with the file such as title, format, and content.
+     *
+     * @return void
+     */
+    public function createFile(ObjectEntity $objectEntity, array $data): File
+    {
+        if (isset($data['versie']) === false || $data['versie'] === null) {
+            $objectEntity->hydrate(['versie' => 1]);
+        }
+
+        if (isset($data['versie']) === true && $data['versie'] !== null) {
+            $objectEntity->hydrate(['versie' => ++$data['versie']]);
+        }
+
+        $file = new File();
+        $file->setBase64('');
+        $file->setMimeType(($data['formaat'] ?? 'application/pdf'));
+        $file->setName($data['titel']);
+        $file->setExtension('');
+        $file->setSize(0);
+
+        return $file;
+
+    }//end createFile()
+
+
+    /**
+     * Creates or updates a file associated with a given ObjectEntity instance.
```

Comment:
Zelfde beschrijving als de createFile functie, wat is het verschil?

Reply:
Moet ik nagaan, deze functies zijn gekopiëerd uit de ZDS bundle